src/wrapped_radix_engine_toolkit/low_level/transaction_manifest/instructions/instructions.rs

```diff
@@ -1,36 +1,79 @@
 use crate::prelude::*;
 
+use radix_engine::types::MANIFEST_SBOR_V1_MAX_DEPTH;
+use radix_engine_toolkit::functions::address::decode as RET_decode_address;
+
 #[derive(Clone, Debug, PartialEq, Eq, derive_more::Display, uniffi::Record)]
 #[display("{}", self.instructions_string())]
 pub struct Instructions {
-    pub(crate) secret_magic: InstructionsSecretMagic, // MUST be first prop, else you break build.
+    secret_magic: InstructionsSecretMagic, // MUST be first prop, else you break build.
     pub network_id: NetworkID,
 }
 
 impl Deref for Instructions {
     type Target = Vec<ScryptoInstruction>;
 
     fn deref(&self) -> &Self::Target {
-        &self.secret_magic.0
+        self.secret_magic.instructions()
     }
 }
 
+#[cfg(test)]
 impl Instructions {
-    pub(crate) fn from_scrypto(
-        instructions: ScryptoInstructions,
+    /// For tests only, does not validate the SBOR depth of the instructions.
+    pub(crate) fn new_unchecked(
+        instructions: Vec<ScryptoInstruction>,
         network_id: NetworkID,
     ) -> Self {
         Self {
-            secret_magic: instructions.into(),
+            secret_magic: InstructionsSecretMagic::new(instructions),
             network_id,
         }
     }
 }
 
+impl Instructions {
+    pub(crate) fn instructions(&self) -> &Vec<ScryptoInstruction> {
+        self.deref()
+    }
+}
+
+impl TryFrom<(&[ScryptoInstruction], NetworkID)> for Instructions {
+    type Error = CommonError;
+
+    fn try_from(
+        value: (&[ScryptoInstruction], NetworkID),
+    ) -> Result<Self, CommonError> {
+        let scrypto = value.0;
+        let network_id = value.1;
+
+        // Verify that the instructions has acceptable depth and are compatible
+        _ = instructions_string_from(scrypto, network_id)?;
```

Comment:
@0xOmarA maybe actually it would be enough to create a DUMMY `NotarizedTransaction` here and compile it? If that works, we know that this `Instructions` can be used... so no *need* (but we still can...to be 100% correct and safe) to compile and validate SBOR length in other types (`TransactionIntent`, `SignedIntent`, `NotarizedIntent`, since they all use `Instructions`. 

What do you think?

Reply:
My personal preference for this would be to keep the check in this way and to have `Instructions` fairly isolated from the other types and not make assumptions about where it will be used. Other types would have their own depth check that would eventually involve checking the depth of the instructions.